airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +279,69 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    @staticmethod
+    def get_tickets(
+        result_return_limit: int, getter: Callable, params: Mapping[str, Any] = None, ticket_paginate_limit: int = 300
+    ) -> Iterator:
+        """
+        Read using getter
+
+        This block extends TicketsAPI Stream to overcome '300 page' server error.
+        Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+        query using 'updated_since' = last_record, if there is more data remaining.
+        """
+        params = params or {}
+        # the maximum page allowed to pull during pagination.
+        # ticket_paginate_limit 300
+        # Start page
+        page = 1
+        # Initial request parameters
+        params = {**params, "per_page": result_return_limit}
+
+        while True:
+            params["page"] = page
+            batch = list(getter(params=params))
+            yield from batch
+
+            if len(batch) < result_return_limit:
+                return iter(())
+
+            # get last_record from latest batch, pos. 0, because of DESC order of records
+            last_record = batch[0]["updated_at"]
+
+            # checkpoint & switch the pagination
+            if page == ticket_paginate_limit:
+                page = 0  # reset page counter
+                # Adding +1 second to last_record value, to avoid record duplication,
+                # this potentialy could lead to skipping some records during the fetch,
+                # in cases where multiple records are created at the same datetime, at the level of seconds
+                # this behaviour is tested in test_300_page.py unit_test (test1)
+                # to avoid such ocations, please comment or remove the `.add(seconds=1)' part.
+                last_record = pendulum.parse(last_record).add(seconds=1)
+                # updating request parameters with last_record state
+                params["order_by"] = "updated_at"
```

Comment:
according to https://developers.freshdesk.com/api/#list_all_tickets  Default sort order is 'created_at'

so we should not specify it at all, or specify it for all tickets requests

Reply:
As for other streams, they use `updated_at`, I didn't change this paradigm, do we need to?

According to[ documentation](https://developers.freshdesk.com/api/#list_all_tickets) we could also use `updated_at` as the sort field equally to created_at - they both are sortable.

Reply:
Fixed, decided to leave the updated_at as default sort and filter option.